code/game/gamemodes/wizard/spellbook.dm

```diff
@@ -47,6 +47,7 @@
 						aspell.name = "Instant [aspell.name]"
 				if(aspell.spell_level >= aspell.level_max)
 					to_chat(user, "<span class='notice'>This spell cannot be strengthened any further.</span>")
+				aspell.create_new_targeting()
```

Comment:
I think it would be better to make a proc on `spell` which will upgrade the spell according to what the spell wants. By default, it'll behave like the above. And you can override it to add new behaviour.

Once you have done that, be sure to set the `targeting` var to this in the `spacetime_dist` spell's override of `upgrade_spell` or however you want to call it.

Reply:
Done!